index.js

```diff
@@ -7,8 +7,8 @@ const parenthesesIncrementer = (filename, extension) => {
 	return `${originalFilename.trim()} (${++index})${extension}`;
 };
 
-const underscoreIncrementer = (filename, extension) => {
-	let [, originalFilename, index] = filename.match(/^(.*)_(\d+)$/) || [null, filename, 0];
+const separatorIncrementer = separator => (filename, extension) => {
+	let [, originalFilename, index] = new RegExp(`^(.*)${separator}(\\d+)$`).exec(filename) || [null, filename, 0];
```

Comment:
You need to escape the separator: https://github.com/sindresorhus/escape-string-regexp